scaletest/dashboard/chromedp.go

```diff
@@ -0,0 +1,216 @@
+package dashboard
+
+import (
+	"context"
+	"math/rand"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/chromedp/cdproto/cdp"
+	"github.com/chromedp/cdproto/network"
+	"github.com/chromedp/chromedp"
+	"golang.org/x/xerrors"
+)
+
+// Action is just a function that does something.
+type Action func(ctx context.Context) error
+
+// Selector locates an element on a page.
+type Selector string
+
+// Target is a thing that can be clicked.
+type Target struct {
+	// Label is a human-readable label for the target.
+	Label Label
+	// ClickOn is the selector that locates the element to be clicked.
+	ClickOn Selector
+	// WaitFor is a selector that is expected to appear after the target is clicked.
+	WaitFor Selector
+}
+
+// Label identifies an action.
+type Label string
+
+var defaultTargets = []Target{
+	{
+		Label:   "workspace_list",
+		ClickOn: `nav a[href="/workspaces"]:not(.active)`,
+		WaitFor: `tr[role="button"][data-testid^="workspace-"]`,
+	},
+	{
+		Label:   "starter_templates",
+		ClickOn: `a[href="/starter-templates"]`,
+		WaitFor: `a[href^="/starter-templates/"]`,
+	},
+	{
+		Label:   "workspace_details",
+		ClickOn: `tr[role="button"][data-testid^="workspace-"]`,
+		WaitFor: `tr[role="button"][data-testid^="build-"]`,
+	},
+	{
+		Label:   "workspace_build_details",
+		ClickOn: `tr[role="button"][data-testid^="build-"]`,
+		WaitFor: `*[aria-label="Build details"]`,
+	},
+	{
+		Label:   "template_list",
+		ClickOn: `nav a[href="/templates"]:not(.active)`,
+		WaitFor: `tr[role="button"][data-testid^="template-"]`,
+	},
+	{
+		Label:   "template_docs",
+		ClickOn: `a[href^="/templates/"][href$="/docs"]:not([aria-current])`,
+		WaitFor: `#readme`,
+	},
+	{
+		Label:   "template_files",
+		ClickOn: `a[href^="/templates/"][href$="/docs"]:not([aria-current])`,
+		WaitFor: `.monaco-editor`,
```

Comment:
nit: should we wrap with an extra `coder-editor` div or `.monaco-editor` introduced by us?

Reply:
I'm not sure, it seems to work for the moment?

Reply:
Ok, let's leave it as is 👍